### DISCUSSION TITLE
DockerOperator does not export context environment variables to container

### DISCUSSION THREAD

**Discussion by jeongukjae** [apache/airflow#24595] [General]
### Apache Airflow Provider(s)

docker

### Versions of Apache Airflow Providers

```
apache-airflow==2.3.2 --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-2.3.2/constraints-3.8.txt"
apache-airflow-providers-docker==3.0.*
```

### Apache Airflow version

2.3.2 (latest released)

### Operating System

MacOS

### Deployment

Virtualenv installation

### Deployment details

`airflow standalone` command. (dev environment)

### What happened

When developing some DAGs in a standalone environment, I couldn't get context environment variables such as `AIRFLOW_CTX_EXECUTION_DATE` in `DockerOperator`.

### What you think should happen instead

I could get the desired env vars in the container.

### How to reproduce

It seems like every docker operator cannot get context env vars.

```python
with DAG(
  pipeline_name,
  description=pipeline_description,
  ...
) as dag:
  DockerOperator(
    task_id="example_task_name",
    image="example_image",
    ...
  )
```

### Anything else

logs:

```
...
logs from taskinstance.py ...
...
[2022-06-22, 15:29:21 UTC] {task_command.py:370} INFO - Running <TaskInstance: ... scheduled__2022-06-22T06:25:02.977109+00:00 map_index=2 [running]> on host 1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa
[2022-06-22, 15:29:21 UTC] {taskinstance.py:1569} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=...
AIRFLOW_CTX_DAG_ID=...
AIRFLOW_CTX_TASK_ID=...
AIRFLOW_CTX_EXECUTION_DATE=2022-06-22T06:25:02.977109+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2022-06-22T06:25:02.977109+00:00
[2022-06-22, 15:29:22 UTC] {docker.py:248} INFO - Starting docker container from image ...
...
Docker container logs ...
...
```

The log said the following env vars are exported, but I cannot get any env var except `AIRFLOW_TMP_DIR `. 

Here's the relevant code.

In `DockerOperator.execute`, I cannot find the statements getting environment variables from the `context` argument. (https://github.com/apache/airflow/blob/providers-docker/3.0.0/airflow/providers/docker/operators/docker.py#L365-L389) By contrast,  in `BashOperator.execute`, there's the line to get the env vars from the `context` argument. (https://github.com/apache/airflow/blob/2.3.2/airflow/operators/bash.py#L184)

It would be great if context environment variables are passed into the containers :)

Thanks!

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by uranusjr** [comment]
The _Exporting the following env vars_ line means the variables are exported to the _operator_, not the _container run by the operator_. To forward those variables to the container, you need to use `environment`.

Maybe we can add a flag on DockerOperator to automatically forward all the `AIRFLOW_*` variables to the container? Not sure, there are implications to blindly forwarding environment variables, the isolation of which is one of the main point to using containers in the first place.

**Reply by potiuk** [reply]
Yeah. It shoudl not happen automatically. There are a host of potential security issues when we do it. I think any such variable should be individually exported when needed (and I belive it is possible even today).

**Comment by bhavaniravi** [comment]
Using `DockerOperator` itself denotes running a container in an environment isolated from core Airflow. From that angle, forwarding the environment variables doesn't feel natural

**Comment by eladkal** [comment]
> Maybe we can add a flag on DockerOperator to automatically forward all the AIRFLOW_* variables to the container? 

I don't think we should. If a specific user wish to have it they can create a custom operator.
I'm converting this issue to GitHub discussion

**Reply by potiuk** [reply]
Agree. This is no-go for me.

**Comment by amitlevy21** [comment]
for future developers, this is how I managed to pass run id to docker environment:
```python
environment = {
    "RUN_ID": "{{ run_id }}",
}

snapshot = DockerOperator(
    task_id="task_id",
    image="image",
    environment=environment,
    command="command",
)
```
